k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -124,23 +129,18 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
               } handle {
                 case _: Reader.ReaderDiscarded =>
               }
-
             })
 
             _processEventStream(
               () => Json.readStream[W](rsp.reader, Api.BufSize),
               resourceVersion
             )
 
-          case http.Status.Gone =>
-            _resourceVersionTooOld()
-
           case status =>
-            close.set(Closable.nop)
+            rsp.reader.discard()
             log.debug("k8s failed to watch resource %s: %d %s", path, status.code, status.reason)
-            val exc = Api.UnexpectedResponse(rsp)
-            state.update(Activity.Failed(exc))
-            Future.exception(exc)
+            val sleep #:: backoffs1 = backoffs0
+            Future.sleep(sleep).before(_resourceVersionTooOld(backoffs1))
```

Comment:
Correct me if I'm wrong, but the `_resourceVersionTooOld` function is now being called due to errors other than just the resource version being too old? 🤷

Reply:
you are correct

Reply:
maybe it needs a different name...